nc/source/random/Random.cpp

```diff
@@ -0,0 +1,52 @@
+#include "random/Random.h"
+
+#include <random>
+
+namespace
+{
+    class FloatDistribution
+    {
+        public:
+            FloatDistribution()
+                : m_device{},
+                  m_gen{m_device()},
+                  m_distribution{-1.0f, std::nextafter(1.0f, std::numeric_limits<float>::max())}
+            {
+            }
+
+        float Generate()
```

Comment:
With the same input does this generate the same output? (is there no behind the scenes seed?) If it does, we should introduce a seed param.